### PR TITLE
Make pill behavior more predictable when scrolling near the top

### PR DESCRIPTION
**Pull request by whoiswillma**
Previously, when the eatery `collectionView` was "bounced" near the top of its bounds, the `EateriesSharedViewConroller` would consider that an upwards scroll then a downwards scroll, causing the pill to hide then appear quickly. By ignoring cases when the `collectionView` is bounced at its top we prevent this.

I also added a timer to delay showing the pill when the eateries `collectionView` stops scrolling.

### CODE REVIEW

File: Eatery/Controllers/Eateries/EateriesSharedViewController.swift [hunk 1]
```diff
@@ -115,23 +117,43 @@ class EateriesSharedViewController: UIViewController {
...
+            adjustedOffset = scrollView.contentOffset.y + scrollView.contentInset.top
+        }
         
-        if lastContentOffset > offset, !pillViewController.isShowingPill {
-            pillViewController.setShowPill(true, animated: true)
-        } else if lastContentOffset < offset, pillViewController.isShowingPill {
-            pillViewController.setShowPill(false, animated: true)
+        if adjustedOffset < 0 {
+            // disregard when the scrollView is "bounced"
+
```
Comment (kevinchan159) [inline Eatery/Controllers/Eateries/EateriesSharedViewController.swift]: delete newline

File: Eatery/Controllers/Eateries/EateriesSharedViewController.swift [hunk 2]
```diff
@@ -115,23 +117,43 @@ class EateriesSharedViewController: UIViewController {
...
+        if adjustedOffset < 0 {
+            // disregard when the scrollView is "bounced"
+            
+            lastContentOffset = 0
+        } else {
+            let isScrollingDownward = adjustedOffset > lastContentOffset
+            
+            if isScrollingDownward, pillViewController.isShowingPill {
+                pillViewController.setShowPill(false, animated: true)
+                print("2")
```
Comment (kevinchan159) [inline Eatery/Controllers/Eateries/EateriesSharedViewController.swift]: delete prints

File: Eatery/Controllers/Eateries/EateriesSharedViewController.swift [hunk 3]
```diff
@@ -115,23 +117,43 @@ class EateriesSharedViewController: UIViewController {
...
+            } else if !isScrollingDownward, !pillViewController.isShowingPill {
+                pillViewController.setShowPill(true, animated: true)
+                print("3")
```
Comment (kevinchan159) [inline Eatery/Controllers/Eateries/EateriesSharedViewController.swift]: delete prints

**Review by kevinchan159 — APPROVED** [review]
lgtm!

**Review by usubzero — APPROVED** [review]

**Review by chalo2000 — COMMENTED** [review]
LGTM, but the same way pill behavior is wonky at the top since it bounces downward, the pill will appear when scrolling too far down because you are bounced upwards. Is it intended behavior to display the pill upon hitting the bottom of the eateries list?

**Pull request merged**